packages/core/http/core-http-server/src/router/route.ts

```diff
@@ -207,6 +207,14 @@ export interface RouteConfigOptions<Method extends RouteMethod> {
    * @remarks This will be surfaced in OAS documentation.
    */
   deprecated?: boolean;
+
+  /**
+   * Release version or date that this route will be removed
+   * Use with `deprecated: true`
+   *
+   * @remarks This will be surfaced in OAS documentation.
+   */
+  'x-discontinued'?: string;
```

Comment:
Nit: What do you think about defining route definitions and kbn/config-schema meta as `{ discontinued: '...' }` instead of `{ 'x-discontinued': '...' }`. Just saves a few keystrokes and slightly awkward string syntax `'my-field-name'`. Then when we render it out we just map it to `x-discontinued`.

Reply:
I prefer consistency with the final output but can add a brief explanation to the field's description.

Reply:
done in https://github.com/elastic/kibana/pull/192331/commits/94bcf57d80f0ae20ba6cbb219d750ac709f1a87e & https://github.com/elastic/kibana/pull/192331/commits/eb701a1b307a95439e63e2c8a077a5e4119050c3